src/hmacKey.ts

```diff
@@ -339,5 +355,57 @@ export class HmacKey extends ServiceObject<HmacKeyMetadata | undefined> {
       baseUrl: `/projects/${projectId}/hmacKeys`,
       methods,
     });
+
+    this.storage = storage;
+    this.instanceRetryValue = storage.retryOptions.autoRetry;
+  }
+
+  /**
+   * Set the metadata for this object.
+   *
+   * @param {object} metadata - The metadata to set on this object.
+   * @param {object=} options - Configuration options.
+   * @param {function=} callback - The callback function.
+   * @param {?error} callback.err - An error returned while making this request.
+   * @param {object} callback.apiResponse - The full API response.
+   */
+  setMetadata(
+    metadata: Metadata,
+    options?: SetMetadataOptions
+  ): Promise<SetMetadataResponse>;
+  setMetadata(metadata: Metadata, callback: MetadataCallback): void;
+  setMetadata(
+    metadata: Metadata,
+    options: SetMetadataOptions,
+    callback: MetadataCallback
+  ): void;
+  setMetadata(
+    metadata: Metadata,
+    optionsOrCallback: SetMetadataOptions | MetadataCallback,
+    cb?: MetadataCallback
+  ): Promise<SetMetadataResponse> | void {
+    // ETag preconditions are not currently supported so retries should be disabled
```

Comment:
unless the retry strategy is `RetryAlways`, right? i made this mistake too in IAM, I will fix it now